packages/webdriverio/src/utils/Timer.ts

```diff
@@ -2,17 +2,21 @@ import { hasWdioSyncSupport, runFnInFiberContext } from '@wdio/utils'
 
 const TIMEOUT_ERROR = 'timeout'
 
-/**
- * Promise-based Timer. Execute fn every tick.
- * When fn is resolved — timer will stop
- * @param {Number} delay - delay between ticks
- * @param {Number} timeout - after that time timer will stop
- * @param {Function} fn - function that returns promise. will execute every tick
- * @param {Boolean} leading - should be function invoked on start
- * @return {promise} Promise-based Timer.
- */
 class Timer {
-    constructor (delay, timeout, fn, leading) {
+    private _delay: number;
+    private _timeout: number;
+    private _fn: () => Promise<boolean> | boolean;
+    private _leading: boolean;
+    private _conditionExecutedCnt: number;
+    private _start?: number;
+    private _ticks?: number;
+    private _timeoutId?: NodeJS.Timeout | null;
+    private _mainTimeoutId?: NodeJS.Timeout | null;
+    private lastError?: Error;
+    private _resolve?: (value?: PromiseLike<Timer>) => void;
+    private _reject?: (reason?: Error) => void;
```

Comment:
Given that we definitely assign these two values in the constructor I would suggest to expect them to exist:
```suggestion
    private _resolve: (value?: PromiseLike<Timer>) => void;
    private _reject: (reason?: Error) => void;
```

If the TS compiler doesn't like it because we assign them within the Promise callback I suggest to have this at the top of the constructor:

```js
const NOOP = () => { throw new Error('not assigned') }

// ...
this._resolve = NOOP
this._reject = NOOP
```